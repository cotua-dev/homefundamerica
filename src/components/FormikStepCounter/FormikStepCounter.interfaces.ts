export interface Props
{
    stepNumber: number;
    totalSteps: number;
}
