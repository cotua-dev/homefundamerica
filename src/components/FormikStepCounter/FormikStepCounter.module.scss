div.stepCounterWrapper {
    text-align: center;
    background-color: #1F72C1;
    color: #FFFFFF;
    font-size: 20px;
    font-weight: 500;
    padding: 10px 32px;
    border-radius: 8px;
    margin-top: 40px;

    @media screen and (max-width: 900px) {
        margin-top: 10px;
        font-size: 14px;
        padding: 10px 20px;
    }
}
