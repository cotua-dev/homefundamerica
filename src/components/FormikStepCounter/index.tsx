import React, { FunctionComponent } from 'react';
import { Props } from './FormikStepCounter.interfaces';
import * as styles from './FormikStepCounter.module.scss';

const FormikStepCounter: FunctionComponent<Props> = (props: Props) =>
{
    return (
        <div className={styles.stepCounterWrapper}>
            {props.stepNumber} of {props.totalSteps}
        </div>
    );
}

export default FormikStepCounter;
