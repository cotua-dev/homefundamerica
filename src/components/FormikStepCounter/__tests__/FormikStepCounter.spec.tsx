import React from 'react';
import { render, RenderResult } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { default as FormikStepCounter } from '..';
import { Props } from '../FormikStepCounter.interfaces';

describe('FormikStepCounter component', (): void => {
    test('renders', (): void => {
        const props: Props = {
            stepNumber: 0,
            totalSteps: 0,
        };
        const { container }: RenderResult = render(<FormikStepCounter {...props}/>);

        expect(container.firstChild).toMatchSnapshot();
    });
});
