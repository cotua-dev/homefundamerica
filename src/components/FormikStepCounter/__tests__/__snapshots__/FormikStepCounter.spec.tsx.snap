// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormikStepCounter component renders 1`] = `
<div>
  0
   of 
  0
</div>
`;
